File: AtCoder/ABC/177/cpp/C_Sum_of_product_of_pairs.cpp
#include <iostream>
#include <stdio.h>

#define MOD 1'000'000'007
#define MAX_N 200'000

using namespace std;
typedef long long ll;

ll N;
ll A[MAX_N];
ll sum[MAX_N + 1];

int main() {
  cin >> N;
  for (int i = 0; i < N; ++i) {
    cin >> A[i];
  }
  for (int i = 0; i < N; ++i) {
    sum[i + 1] = (sum[i] + A[i]) % MOD;
  }
  ll res = 0;
  for (int i = 0; i < N - 1; ++i) {
    res = (res + A[i] * ((sum[N] - sum[i + 1] + MOD) % MOD)) % MOD;
  }
  printf("%lld\n", res);
}

File: AtCoder/ABC/177/python/C_Sum_of_product_of_pairs.py
MOD = int(1e9) + 7
N = int(input())
A = list(map(int, input().split()))

cum = [0] * (N + 1)
for i in range(N):
    cum[i + 1] = (cum[i] + A[i]) % MOD
res = 0
for i in range(N - 1):
    res = (res + A[i] * (cum[N] - cum[i + 1])) % MOD
res = (res + MOD) % MOD
print(res)
